src/status_im/chat/models/message.cljs

```diff
@@ -211,6 +212,25 @@
                        (add-message chat-id message-with-id true)
                        (send chat-id send-record))))
 
+(defn update-message-status [{:keys [chat-id message-id from] :as message} status {:keys [db]}]
+  (let [updated-message (assoc-in message [:user-statuses from] status)]
+    {:db                        (assoc-in db [:chats chat-id :messages message-id] updated-message)
+     :data-store/update-message updated-message}))
+
+(defn resend-message [chat-id message-id cofx]
```

Comment:
Worth noting that this message might not appear as the last message in the chat, unless we update the clock value.

Basically say you have received `B` & `C` after trying to send `A`. You try to send `A` again, should `A` be displayed before or after `B` & `C`? Current codebase would display it before, is that what we want?

Not sure what is the desired behaviour myself, sorry just bringing problems to the table :)

Reply:
@cammellos i would display it before, it preserves ordering in UI and user won't be surprised why message has been moved on Resend.

Reply:
@dmitryn fair!